api_tests/tests/conftest.py

```diff
@@ -64,17 +67,122 @@ def helper():
     return GenericRequest()
 
 
-@pytest.fixture(autouse=True)
-def setup(request):
-    """This function is called before each test is executed"""
-    # Get the name of the current test and attach it the the test instance
-    name = (request.node.name, request.node.originalname)[request.node.originalname is not None]
-    setattr(request.cls, "name", name)
+def _set_default_rate_limit(product: ApigeeApiProducts):
+    product.update_ratelimits(quota=60000,
+                              quota_interval="1",
+                              quota_time_unit="minute",
+                              rate_limit="1000ps")
+
+
+@pytest.fixture()
+async def test_product():
+    """Create a test product which can be modified by the test"""
+    product = ApigeeApiProducts()
+    await product.create_new_product()
+    _set_default_rate_limit(product)
+    yield product
+    await product.destroy_product()
+
+
+@pytest.fixture()
+def app():
+    return ApigeeApiDeveloperApps()
+
+
+@pytest.fixture()
+async def test_app(app):
+    """Create a test app which can be modified in the test"""
+    await app.create_new_app()
+
+    yield app
+    await app.destroy_app()
 
-    oauth = OauthHelper(config.CLIENT_ID, config.CLIENT_SECRET, config.REDIRECT_URI)
-    setattr(request.cls, "oauth", oauth)
 
-    yield  # Handover to test
+@pytest.yield_fixture(scope="class")
+def setup_user_restricted_app(request, test_app, test_product):
+    print("\nCreating User Restricted App..")
+    asyncio.run(test_app.create_new_app())
```

Comment:
Could this setup be simplified a bit by making it async instead of sync?

Reply:
i don't think you can make a class fixture async.

Reply:
I have removed this bit of code as it not currently needed/used